//===- Block.proto ----------------------------------------------*- C++ -*-===//
//
//  Copyright (C) 2018 GrammaTech, Inc.
//
//  This code is licensed under the MIT license. See the LICENSE file in the
//  project root for license terms.
//
//  This project is sponsored by the Office of Naval Research, One Liberty
//  Center, 875 N. Randolph Street, Arlington, VA 22203 under contract #
//  N68335-17-C-0700.  The content of the information does not necessarily
//  reflect the position or policy of the Government and no official
//  endorsement should be inferred.
//
//===----------------------------------------------------------------------===//
syntax = "proto3";
package proto;

enum Exit {
    Fallthrough = 0;
    Branch = 1;
    Call = 2;
    Return = 3;
}

message Block {
    bytes uuid = 1;
    uint64 address = 2;
    uint64 size = 3;
    uint64 decode_mode = 4;
    Exit exit_kind = 5;
}
